src/Microsoft.Azure.WebJobs.Extensions.Kafka/Diagnostics/BatchEventActivityProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    internal class BatchEventActivityProvider : ActivityProvider, IDisposable
+    {
+        private List<ActivityLink> activityLinks;
+        private readonly IKafkaEventData[] kafkaEvents;
+        private static string KafkaBatchTriggerActivityName { get; } = "MultipleKafkaTrigger.Process";
+
+        public BatchEventActivityProvider(IKafkaEventData[] kafkaEvents, string consumerGroup) : base(GetTopic(kafkaEvents), consumerGroup)
+        {
+            this.kafkaEvents = kafkaEvents;
+            this.activityLinks = new List<ActivityLink>();
+            this.CreateActivity();
+        }
+
+        private static string GetTopic(IKafkaEventData[] kafkaEvents)
+        {
+            if (kafkaEvents.Length == 0)
+            {
+                throw new Exception("KafkaEvents array is null");
+            }
+            return kafkaEvents[0].Topic;
+        }
+
+        public void CreateActivity()
+        {
+            if (KafkaActivitySource.HasListeners())
```

Comment:
if the else case is not there, the recommendation would be to make :-
            if (!KafkaActivitySource.HasListeners()) return;

Reply:
When the if condition fails, it will automatically return.